src/redist/targets/GenerateMSIs.targets

```diff
@@ -127,28 +135,61 @@
                       -InformationAction Continue" />
   </Target>
 
-  <Target Name="GenerateTemplatesMsi"
-        DependsOnTargets="GenerateLayout;AcquireWix;MsiTargetsSetupInputOutputs;SetSdkBrandingInfo"
-        Condition=" '$(OS)' == 'Windows_NT' "
-        Inputs="$(TemplatesLayoutPath)**/*;
-                  $(TemplatesGenerateMsiPowershellScript)"
-        Outputs="$(TemplatesMSIInstallerFile)">
+  <Target Name="GenerateTemplatesMsis"
+          DependsOnTargets="GenerateLayout;AcquireWix;MsiTargetsSetupInputOutputs;SetSdkBrandingInfo;SetupTemplatesMsis"
+          Condition="$(ProductMonikerRid.StartsWith('win')) And !$(Architecture.StartsWith('arm'))"
+          Inputs="@(TemplatesMsiComponent->'%(LayoutPath)');
+                    $(TemplatesGenerateMsiPowershellScript)"
+          Outputs="@(TemplatesMsiComponent->'%(MSIInstallerFile)')">
 
     <Exec Command="powershell -NoProfile -NoLogo $(TemplatesGenerateMsiPowershellScript) ^
-                      '$(TemplatesLayoutPath.TrimEnd('\'))' ^
-                      '$(TemplatesMSIInstallerFile)' ^
+                      '@(TemplatesMsiComponent->'%(LayoutPath)')' ^
+                      '%(TemplatesMsiComponent.MSIInstallerFile)' ^
                       '$(WixRoot)' ^
-                      '$(SdkBrandName)' ^
-                      '$(MsiVersion)' ^
-                      '$(SDKBundleVersion)' ^
-                      '$(NugetVersion)' ^
-                      '$(TemplatesInstallerUpgradeCode)' ^
+                      '@(TemplatesMsiComponent->'%(BrandName)')' ^
+                      '@(TemplatesMsiComponent->'%(MsiVersion)')' ^
+                      '@(TemplatesMsiComponent->'%(MsiVersion)')' ^
+                      '@(TemplatesMsiComponent->'%(Version)')' ^
+                      '@(TemplatesMsiComponent->'%(UpgradeCode)')' ^
                       '$(Architecture)' ^
                       -InformationAction Continue" />
   </Target>
 
+  <Target Name="SetupTemplatesMsis">
+    <PropertyGroup>
+      <TemplatesGenerateMsiPowershellScript>$(MSBuildThisFileDirectory)packaging/windows/clisdk/generatetemplatesmsi.ps1</TemplatesGenerateMsiPowershellScript>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <TemplatesMsiComponent Include="NetCore30TemplatesMsi">
+        <LayoutPath>$(Templates30LayoutPath.TrimEnd('\'))</LayoutPath>
+        <MSIInstallerFile>$(Templates30MSIInstallerFile)</MSIInstallerFile>
+        <BrandName>Microsoft .NET Core 3.0 Templates</BrandName>
```

Comment:
Should these be here or in the branding.targets?

Reply:
Depends. If we are still thinking about moving the templates manufactoring to a 'templates' repo, we should keep eveything centralized. I'll go ahead and move these into branding.targets because the templates manufacturing will be in core-sdk for quite some time.